src/nfa/nfa-concatenation.cc

```diff
@@ -70,37 +80,72 @@ class Concatenation
     Nfa result{}; ///< Concatenated automaton.
     StateToStateMap lhs_result_states_map{}; ///< Map mapping @c lhs states to @c result states.
     StateToStateMap rhs_result_states_map{}; ///< Map mapping @c rhs states to @c result states.
+    Symbol epsilon{}; ///< Symbol to use as an epsilon symbol to concatenate with.
 
     /**
      * Compute concatenation of given automata.
      */
     void compute_concatenation()
     {
+        if (lhs.initialstates.empty() || lhs.finalstates.empty() || rhs.initialstates.empty()) { return; }
+        const size_t result_num_of_states(lhs_states_num - lhs.finalstates.size() + rhs_states_num);
+        if (result_num_of_states == 0) { return; }
+        result.increase_size(result_num_of_states);
         map_states_to_result_states();
         make_initial_states();
         add_lhs_transitions();
         make_final_states();
         add_rhs_transitions();
     }
 
-    /**'
+    /**
+     * Compute concatenation of given automata concatenating over epsilon transitions.
+     */
+    void compute_epsilon_concatenation() {
```

Comment:
Maybe rename to `compute_concatenation_with_epsilon` or `compute_concatenation_with_epsilon_transitions`?

Reply:
The best name exactly describing the operation is `compute_concatenation_over_epsilon_transitions`. What do you say?

Reply:
Sounds good.

Reply:
I think it's too loooooooong. I like the original version more, however this is probably better understandable so even someone after braing surgery would know what function does.

Reply:
Or maybe just `compute_concatenation_over_epsilon`?

Reply:
or `concatenate_over_epsilon`?

Reply:
No issues with thtat

Reply:
`concatenate_over_epsilon` can be used, too.